sdks/builds/runtime.mk

```diff
@@ -142,7 +142,7 @@ _cross-runtime_$(1)_CONFIGURE_FLAGS= \
 
 .stamp-$(1)-$$(CONFIGURATION)-configure: | $$(if $$(IGNORE_PROVISION_LLVM),,provision-$(5))
 
-$$(TOP)/sdks/builds/$(1)-$$(CONFIGURATION)/$(3).h: .stamp-$(4)-$$(CONFIGURATION)-configure .stamp-$(1)-$$(CONFIGURATION)-configure $$(TOP)/tools/offsets-tool/MonoAotOffsetsDumper.exe | configure-$(4)
+$$(TOP)/sdks/builds/$(1)-$$(CONFIGURATION)/$(3).h: .stamp-$(4)-configure .stamp-$(1)-$$(CONFIGURATION)-configure $$(TOP)/tools/offsets-tool/MonoAotOffsetsDumper.exe | configure-$(4)
```

Comment:
Well, for some reason `configure-$(4)` is not doing what's supposed to because without this extra dependency `make configure-wasm-cross-win` fails because `wasm-runtime-release` was not configured.

This was after applying all the other changes you proposed.